### PR TITLE
Updated setting of client options to allow passthrough.

### PR DESCRIPTION
**Pull request by GROwen**
### Changes
1. Updating the setting of the ApiClient options to handle configurable timeout.

I've tested this using the changes in this CMDB PR https://github.com/dpc-sdp/sdp-cmdb/pull/599, which I'm going to leave in draft until this work is merged.

### CODE REVIEW

**Comment by GROwen** [comment]
👋 @steveworley @yusufhm I've limited this update to the variable plugins because it's where it will have most impact.

Some of the other [plugins/client methods like project_deploy](https://github.com/salsadigitalauorg/lagoon_ansible_collection/blob/28236a0a3bc32440e274898fde859cd74a773462/api/plugins/action/deploy.py#L30) will need more of a refactor so that existing args use the options dict instead of individual args. i.e. wait, delay and retries would be a part of the options dict instead of individual args on the task.

File: api/plugins/action/env_variable.py [hunk 1]
```diff
@@ -22,19 +22,20 @@ def run(self, tmp=None, task_vars=None):
...
-
+        name = self._task.args.get('name')
         type = self._task.args.get('type')
         type_name = self._task.args.get('type_name')
-        name = self._task.args.get('name')
         state = self._task.args.get('state', 'present')
         value = self._task.args.get('value', None)
         scope = self._task.args.get('scope', None)
         replace_existing = self._task.args.get('replace_existing', False)
+        options = self._task.args.get('options', {'headers': {}})
```
Comment (steveworley) [inline api/plugins/action/env_variable.py:32]:
```suggestion
        options = self._task.args.get('options', {}})
        if not 'headers' in options:
            options['headers'] = {}
```
I think we should always default headers to `{}` to not replace current behaviour

File: api/plugins/action/variables.py [hunk 2]
```diff
@@ -22,15 +22,16 @@ def run(self, tmp=None, task_vars=None):
 
         display.v("Task args: %s" % self._task.args)
 
+        name = self._task.args.get('name')
+        type = self._task.args.get('type', 'project')
+        options = self._task.args.get('options', {'headers': {}})
```
Comment (steveworley) [inline api/plugins/action/variables.py:27]: Same as above :)

**Review by steveworley — CHANGES_REQUESTED** [review]
Minor change :)

**Review by steveworley — APPROVED** [review]

**Pull request merged**